app/controllers/case_groups_controller.rb

```diff
@@ -24,27 +28,30 @@ def create
   end
 
   def update
-    @case_group = current_organization.case_groups.find(params[:id])
+    authorize @case_group
+
     if @case_group.update(case_group_params)
       redirect_to case_groups_path, notice: "Case group updated!"
     else
-      render :new, status: :unprocessable_entity
+      render :edit, status: :unprocessable_entity
     end
   end
 
   def destroy
-    case_group = current_organization.case_groups.find(params[:id])
-    case_group.destroy
+    authorize @case_group
+
+    @case_group.destroy
     redirect_to case_groups_path, notice: "Case group deleted!"
   end
 
   private
 
   def case_group_params
+    params.merge(casa_org: current_organization)
     params.require(:case_group).permit(:name, casa_case_ids: [])
   end
 
-  def authorize_admin_or_supervisor!
-    authorize :application, :admin_or_supervisor?
+  def set_case_group
+    @case_group = CaseGroup.find(params[:id])
```

Comment:
Some controllers use `policy_scope(CaseGroup).find(params[:id])` here. That makes sense, but raises an ActiveRecord not found exception if an unauthorized user... We handle authorization failures in ApplicationController with a redirect to root_path (`rescue_from Pundit::NotAuthorizedError, with: :not_authorized`). I think I prefer to let that happen.

Reply:
Thinking about this some more and I think probably better to use policy_scope and let the unauthorized user think that the record doesn't exist - don't leak information.